src/include/duckdb/storage/serialization/parsed_expression.json

```diff
@@ -0,0 +1,376 @@
+    "base": "ParsedExpression",
+    "enum": "DEFAULT",
+    "members": [
+    ],
+    "constructor": []
+  },
+  {
+    "class": "FunctionExpression",
+    "base": "ParsedExpression",
+    "enum": "FUNCTION",
+    "members": [
+      {
+        "name": "function_name",
+        "type": "string"
+      },
+      {
+        "name": "schema",
+        "type": "string"
+      },
+      {
+        "name": "children",
+        "type": "vector<ParsedExpression*>"
+      },
+      {
+        "name": "filter",
+        "optional": true,
+        "type": "ParsedExpression*"
+      },
+      {
+        "name": "order_bys",
+        "type": "OrderModifier*",
+        "base": "ResultModifier"
+      },
+      {
+        "name": "distinct",
+        "type": "bool"
+      },
+      {
+        "name": "is_operator",
+        "type": "bool"
+      },
+      {
+        "name": "export_state",
+        "type": "bool"
+      },
+      {
+        "name": "catalog",
+        "type": "string"
+      }
+    ],
+    "constructor": []
+  },
+  {
+    "class": "LambdaExpression",
+    "base": "ParsedExpression",
+    "enum": "LAMBDA",
+    "members": [
+      {
+        "name": "lhs",
+        "type": "ParsedExpression*"
+      },
+      {
+        "name": "expr",
+        "type": "ParsedExpression*"
+      }
+    ],
+    "constructor": []
+  },
+  {
+    "class": "OperatorExpression",
+    "base": "ParsedExpression",
+    "enum": "OPERATOR",
+    "members": [
+      {
+        "name": "children",
+        "type": "vector<ParsedExpression*>"
+      }
+    ]
+  },
+  {
+    "class": "ParameterExpression",
+    "base": "ParsedExpression",
+    "enum": "PARAMETER",
+    "members": [
+      {
+        "name": "parameter_nr",
+        "type": "idx_t"
+      }
+    ],
+    "constructor": []
+  },
+  {
+    "class": "PositionalReferenceExpression",
+    "base": "ParsedExpression",
+    "enum": "POSITIONAL_REFERENCE",
+    "members": [
+      {
+        "name": "index",
+        "type": "idx_t"
+      }
+    ],
+    "constructor": []
+  },
+  {
+    "class": "StarExpression",
+    "base": "ParsedExpression",
+    "enum": "STAR",
+    "members": [
+      {
+        "name": "relation_name",
+        "type": "string"
+      },
+      {
+        "name": "exclude_list",
+        "type": "unordered_set<string>"
+      },
+      {
+        "name": "replace_list",
+        "type": "unordered_set<ParsedExpression*>"
+      },
+      {
+        "name": "columns",
+        "type": "bool"
+      },
+      {
+        "name": "expr",
+        "optional": true,
+        "type": "ParsedExpression*"
+      }
+    ],
+    "constructor": []
+  },
+  {
+    "class": "SubqueryExpression",
+    "base": "ParsedExpression",
+    "enum": "SUBQUERY",
+    "members": [
+      {
+        "name": "subquery_type",
+        "type": "SubqueryType"
+      },
+      {
+        "name": "subquery",
+        "type": "SelectStatement"
+      },
+      {
+        "name": "child",
+        "optional": true,
```

Comment:
Just a comment, I would potentially consider in a next step to remove 'optional' and move that as part of the type, so here for example:
```
"name": "child",
"type": "duckdb::optional<ParsedExpression*>"
```
with the appropriate serialization / deserialization deducted from the type instead than from the extra field, but I am not sure what's more expensive to maintain in the long run.

Also here likely to be re-evaluated later.